### ISSUE TITLE
Severe crashes when loading fbx models!

### ISSUE THREAD

**Issue by SuperYF-86** [helix-toolkit/helix-toolkit#1909]
[GPU: 1080ti 11G and latest version of drivers].
The program crashed when loading some fbx models that contained a lot of textures. This happens a lot, but of course there are cases where the loading succeeds.

At best VisualStudio catches the exception, but sometimes the program just quits, at worst windows gets stuck and I can't move the mouse or open task manager. There was nothing I could do except cut the power, which was really scary.

I have observed that as the file size increases, the success rate of loading fbx models with lots of textures decreases. There are no exceptions when the file is less than 30mb, but if the file is over 150mb it is almost impossible to load and often has unpredictable and serious consequences (PC shock).
![ex_1](https://user-images.githubusercontent.com/72618639/221661855-5552177c-547e-42f3-8a0e-61498c58f856.png)

Also, I have a couple of exceptions to report.
1, CrossSectionMeshGeometryModel3D does not show shadows.
2, the last version had a nice refresh rate optimisation for SwapChainRendering, but it didn't work every time.
I found it possible to make the WPF animation loop infinitely to ensure that the WPF application refreshes aggressively, but I'm not sure if this makes sense.

**Comment by holance** [comment]
Seems like the graphics card driver crashed. Are you running the program on laptop with integrated graphics card?

**Comment by YOYOyayaLaLaNiNi** [comment]
> Seems like the graphics card driver crashed. Are you running the program on laptop with integrated graphics card?

No, my machine is a desktop PC and has integrated graphics blocked. I forgot to mention that this problem only occurs in an asynchronous loading environment.

**Comment by holance** [comment]
Could you provide sample code on async loading?

**Comment by YOYOyayaLaLaNiNi** [comment]
> Could you provide sample code on async loading? Does the file load demo from example crash as well?



The exception is reproduced in [WPF]FileLoadDemo, but never in [WinUI]ModelViewer.

Unfortunately, although this is a necessary way to solve the problem, I am not allowed to make the code public at this stage.

**Comment by holance** [comment]
Could you provide the model?

**Comment by YOYOyayaLaLaNiNi** [comment]
> Could you provide the model?


The test files I used were all sourced from GoogleEarth and exported as fbx via RenderDoc+Blender. these files have some commonalities: broken meshes and multiple Textures.

I found that only such fbx files larger than 70mb caused exceptions in asynchronous loading, and I was unable to upload files larger than 25bm here.

**Comment by holance** [comment]
You can create a repo and upload it
